src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Security/DoNotDisableTokenValidationChecks.cs

```diff
@@ -0,0 +1,91 @@
+        private static readonly LocalizableString s_Title = new LocalizableResourceString(
+            nameof(MicrosoftNetCoreAnalyzersResources.DoNotDisableTokenValidationChecks),
+            MicrosoftNetCoreAnalyzersResources.ResourceManager,
+            typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_Message = new LocalizableResourceString(
+            nameof(MicrosoftNetCoreAnalyzersResources.DoNotDisableTokenValidationChecksMessage),
+            MicrosoftNetCoreAnalyzersResources.ResourceManager,
+            typeof(MicrosoftNetCoreAnalyzersResources));
+        private static readonly LocalizableString s_Description = new LocalizableResourceString(
+            nameof(MicrosoftNetCoreAnalyzersResources.DoNotDisableTokenValidationChecksDescription),
+            MicrosoftNetCoreAnalyzersResources.ResourceManager,
+            typeof(MicrosoftNetCoreAnalyzersResources));
+
+        internal static DiagnosticDescriptor Rule = DiagnosticDescriptorHelper.Create(
+                DiagnosticId,
+                s_Title,
+                s_Message,
+                DiagnosticCategory.Security,
+                RuleLevel.BuildWarning,
+                description: s_Description,
+                isPortedFxCopRule: false,
+                isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+
+            context.RegisterCompilationStartAction(compilationStartAnalysisContext =>
+            {
+                var compilation = compilationStartAnalysisContext.Compilation;
+                var tokenValidationParamsTypeSymbol = compilation.GetOrCreateTypeByMetadataName(
+                    WellKnownTypeNames.MicrosoftIdentityModelTokensTokenValidationParameters);
+
+                if (tokenValidationParamsTypeSymbol == null)
+                {
+                    return;
+                }
+
+                compilationStartAnalysisContext.RegisterOperationAction(operationAnalysisContext =>
+                {
+                    var simpleAssignmentOperation = (ISimpleAssignmentOperation)operationAnalysisContext.Operation;
+
+                    if (simpleAssignmentOperation.Target is IPropertyReferenceOperation propertyReferenceOperation)
+                    {
+                        var property = propertyReferenceOperation.Property;
+
+                        if (property.ContainingType.Equals(tokenValidationParamsTypeSymbol) &&
```

Comment:
```suggestion
                        if (property.ContainingType.Equals(tokenValidationParamsTypeSymbol, SymbolEqualityComparer.Default) &&
```

Reply:
Looks like this overload doesn't exist in 2.9.x